### PR TITLE
docs: Update migration-v4 doc

### LINKED ISSUE #21373: table组件在dataIndex属性使用多层JSON无效
- [x] I have searched the [issues](https://github.com/ant-design/ant-design/issues) of this repository and believe that this is not a duplicate.

### Reproduction link
[Edit on CodeSandbox](https://codesandbox.io/s/exciting-glade-2pzk0?fontsize=14&hidenavigation=1&theme=dark)

### Steps to reproduce
性别这一列期望能正常显示

### What is expected?
性别这一列根据数据显示

### What is actually happening?
不显示

| Environment | Info |
|---|---|
| antd | 4.0.0-rc.4 |
| React | 16.12 |
| System | win10 |
| Browser | chrome 80 |

---
antd3.x版本可以

### PR DESCRIPTION
**Pull request by zombieJ**
### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [x] Site / document update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 🔗 Related issue link

close #21373

### 💡 Background and solution

-----
[View rendered docs/react/migration-v4.zh-CN.md](https://github.com/ant-design/ant-design/blob/v4-docs/docs/react/migration-v4.zh-CN.md)

### CODE REVIEW

File: docs/react/migration-v4.zh-CN.md [hunk 1]
```diff
@@ -117,7 +120,9 @@ const Demo = () => (
 - Grid 组件使用 flex 布局。
 - Button 的 `danger` 现在作为一个属性而不是按钮类型。
 - Input、Select 的 `value` 为 `undefined` 时改为非受控状态。
-- Table 在没有 `columns` 时仍然会保留一列。
+- Table 重写
+  - 在没有 `columns` 时仍然会保留一列。
+  - 嵌套 `dataIndex` 支持从 `'xxx.yyy'` 改成 `['xxx', 'yyy']`。
```
Comment (afc163) [inline docs/react/migration-v4.zh-CN.md]: 给个 diff 代码吧。

File: docs/react/migration-v4.en-US.md [hunk 2]
```diff
@@ -117,7 +120,20 @@ const Demo = () => (
...
+```diff
+<Table
+  columns={[
+    {
+      title: 'Age',
+---      dataIndex: 'user.age',
++++      dataIndex: ['user', 'age'],
+    },
+  ]}
+/>
```
Comment (afc163) [inline docs/react/migration-v4.en-US.md]: - 注意缩进。
- 一个 `-` 和 `+` 就可以了。

File: docs/react/migration-v4.zh-CN.md [hunk 3]
```diff
@@ -108,7 +108,10 @@ const Demo = () => (
 
 #### 组件重构
 
-- Form 重写，不再需要 `Form.create`，迁移文档请查看[此处](/components/form/v3)。
+- Form 重写
+  - 不再需要 `Form.create`。
+  - 嵌套字段支持从 `'xxx.yyy'` 改成 `['xxx', 'yyy']`。
+  - 迁移文档请查看[此处](/components/form/v3)。
```
Comment (afc163) [inline docs/react/migration-v4.zh-CN.md]:
```suggestion
  - 迁移文档请查看[此处](/components/form-cn/v3)。
```

**Comment by afc163** [comment]
image

缩进不对。

**Comment by afc163** [comment]
image

上面这里也不对。

File: docs/react/migration-v4.zh-CN.md [hunk 4]
```diff
@@ -117,7 +120,21 @@ const Demo = () => (
...
+<Table
+  columns={[
+    {
+      title: 'Age',
+-       dataIndex: 'user.age',
++       dataIndex: ['user', 'age'],
+    },
+  ]}
+/>
+```
```
Comment (afc163) [inline docs/react/migration-v4.zh-CN.md]: 这个还是没改。

Comment (zombieJ) [reply docs/react/migration-v4.zh-CN.md]: 我二了，原来是同级的

**Review by afc163 — APPROVED** [review]

**Comment by zombieJ** [comment]
想起来内部链接不用加 `-cn`，它是自动切换的。

**Pull request merged**